src/NuGet.Core/NuGet.Packaging/Signing/Utility/CertificateUtility.cs

```diff
@@ -307,5 +308,17 @@ public static bool IsSelfIssued(X509Certificate2 certificate)
                 return true;
             }
         }
+
+        public static IReadOnlyList<byte[]> GetRawDataForCollection(X509Certificate2Collection certificates)
+        {
+            var certificatesRawData = new List<byte[]>();
```

Comment:
You know exactly how many elements there will be so:

`var certificatesRawData = new List<byte[]>(capacity: certificates.Count);`